src/runtime/containerd-shim-v2/create.go

```diff
@@ -259,3 +272,99 @@ func doMount(mounts []*containerd_types.Mount, rootfs string) error {
 	}
 	return nil
 }
+
+func configureNonRootHypervisor(runtimeConfig *oci.RuntimeConfig) error {
+	userName, err := createVmmUser()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			removeVmmUser(userName)
+		}
+	}()
+
+	u, err := user.Lookup(userName)
+	if err != nil {
+		return err
+	}
+
+	uid, err := strconv.Atoi(u.Uid)
+	if err != nil {
+		return err
+	}
+	gid, err := strconv.Atoi(u.Gid)
+	if err != nil {
+		return err
+	}
+	runtimeConfig.HypervisorConfig.Uid = uint32(uid)
+	runtimeConfig.HypervisorConfig.Gid = uint32(gid)
+
+	userTmpDir := path.Join("/run/user/", fmt.Sprint(uid))
```

Comment:
I have some concerns that we are now leaving artifacts at yet another location on the filesystem.

Today, we leave artifacts for a running pod at:
/run/vc/
/run/kata-containers

As far as I understand, instead of leaving artifacts at `/run/vc/vm/`, we'll now leave them at `/run/user/<user id>/` when using this configuration?

In a perfect world, we'd have everything under `/run/kata-containers/` ...

Is the /run/user/<> part required in order to ensure appropriate permissions?

Reply:
> Is the /run/user/<> part required in order to ensure appropriate permissions?

That's correct. Both `/run/vc` and `/runc/kata-containers` are only accessible by root (probably for good reasons).